backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java

```diff
@@ -0,0 +1,666 @@
+      if (ownerRoleBindingName != null) {
+        roleBindingNames.add(ownerRoleBindingName);
+      }
+
+      if (project.getProject_visibility() == ProjectVisibility.PUBLIC.getNumber()) {
+        String publicReadRoleBindingName =
+            roleService.buildRoleBindingName(
+                ModelDBConstants.ROLE_PROJECT_PUBLIC_READ,
+                projectId,
+                authService.getVertaIdFromUserInfo(unsignedUser),
+                ModelResourceEnum.ModelDBServiceResourceTypes.PROJECT.name());
+        if (publicReadRoleBindingName != null) {
+          roleBindingNames.add(publicReadRoleBindingName);
+        }
+      }
+
+      // Delete workspace based roleBindings
+      List<String> workspaceRoleBindingNames =
+          roleService.getWorkspaceRoleBindings(
+              project.getWorkspace(),
+              WorkspaceTypeEnum.WorkspaceType.forNumber(project.getWorkspace_type()),
+              projectId,
+              ModelDBConstants.ROLE_PROJECT_ADMIN,
+              ModelResourceEnum.ModelDBServiceResourceTypes.PROJECT,
+              ProjectVisibility.forNumber(project.getProject_visibility())
+                  .equals(ProjectVisibility.ORG_SCOPED_PUBLIC),
+              "_GLOBAL_SHARING");
+      roleBindingNames.addAll(workspaceRoleBindingNames);
+    }
+
+    roleService.deleteAllResources(
+        allowedProjects.stream().map(ProjectEntity::getId).collect(Collectors.toList()),
+        ModelResourceEnum.ModelDBServiceResourceTypes.PROJECT);
+  }
+
+  private void deleteExperiments(Session session) {
+    LOGGER.debug("Experiment deleting");
+    String deleteExperimentQueryString =
+        new StringBuilder("FROM ")
+            .append(ExperimentEntity.class.getSimpleName())
+            .append(" ex WHERE ex.")
+            .append(ModelDBConstants.DELETED)
+            .append(" = :deleted ")
+            .toString();
+
+    Query experimentDeleteQuery = session.createQuery(deleteExperimentQueryString);
+    experimentDeleteQuery.setParameter("deleted", true);
+    experimentDeleteQuery.setMaxResults(this.recordUpdateLimit);
+    LOGGER.debug("Experiment delete query: {}", experimentDeleteQuery.getQueryString());
+    List<ExperimentEntity> experimentEntities = experimentDeleteQuery.list();
+
+    List<String> experimentIds = new ArrayList<>();
+    if (!experimentEntities.isEmpty()) {
+      List<String> roleBindingNames = new LinkedList<>();
+      for (ExperimentEntity experimentEntity : experimentEntities) {
+        experimentIds.add(experimentEntity.getId());
+        String ownerRoleBindingName =
+            roleService.buildRoleBindingName(
+                ModelDBConstants.ROLE_EXPERIMENT_OWNER,
+                experimentEntity.getId(),
+                experimentEntity.getOwner(),
+                ModelResourceEnum.ModelDBServiceResourceTypes.EXPERIMENT.name());
+        if (ownerRoleBindingName != null) {
+          roleBindingNames.add(ownerRoleBindingName);
+        }
```

Comment:
Be consistent. This is a separate function for projects, so make it a separate function for experiments too.

Reply:
Created separate functions called `deleteROleBindingsFor*` and moved all delete role bindings code in that function for each entity.